spimple/utils.py

```diff
@@ -157,15 +165,19 @@ def convolve2gaussres(image, xx, yy, gaussparf, nthreads, gausspari=None, pfrac=
     imhat = r2c(iFs(image, axes=ax), axes=ax, forward=True, nthreads=nthreads, inorm=0)
 
     # convolve to desired resolution
-    if gausspari is None:
+    if gausspari in [None, ()]:
         imhat *= gausskernhat
     else:
         for i in range(nband):
             thiskern = Gaussian2D(xx, yy, gausspari[i], normalise=norm_kernel).astype(np.float64)
             thiskern = np.pad(thiskern[None], padding, mode='constant')
             thiskernhat = r2c(iFs(thiskern, axes=ax), axes=ax, forward=True, nthreads=nthreads, inorm=0)
 
-            convkernhat = np.where(np.abs(thiskernhat)>1e-10, gausskernhat/thiskernhat, 0.0)
+            if not np.all(np.isnan(thiskernhat)):
```

Comment:
I suspect this is related to the comments above but the kernel shouldn't ever consist of nans so would prefer a cleaner solution